litgpt/generate/base.py

```diff
@@ -49,7 +49,9 @@ def sample(
         # do not use `torch.where` as in nanogpt because it will repeat top-k collisions
         logits = torch.full_like(logits, float("-inf")).scatter_(-1, i, v)
     # optionally scale the logits and sample from a probability distribution
-    if temperature > 0.0 and top_p is not None and top_p > 0.0:
+    if top_p is None:
```

Comment:
In that case, we could also just set it to 1.0 as the default. Any preference @carmocca ?

Reply:
Yeah, I could set the default to `1.0` and check that is `0.0 <= top_p <= 1.0` and raise and error if it's not the case

Reply:
Agreed!